helper/logging/logging_http_transport.go

```diff
@@ -0,0 +1,279 @@
+	// and NewSubsystemLoggingHTTPTransport when logging the type of HTTP operation via tflog.
+	FieldHttpOperationType = "tf_http_op_type"
+
+	// OperationHttpRequest is the field value used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request via tflog.
+	OperationHttpRequest = "request"
+
+	// OperationHttpResponse is the field value used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response via tflog.
+	OperationHttpResponse = "response"
+
+	// FieldHttpRequestMethod is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request method via tflog.
+	FieldHttpRequestMethod = "tf_http_req_method"
+
+	// FieldHttpRequestUri is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request URI via tflog.
+	FieldHttpRequestUri = "tf_http_req_uri"
+
+	// FieldHttpRequestProtoVersion is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request HTTP version via tflog.
+	FieldHttpRequestProtoVersion = "tf_http_req_version"
+
+	// FieldHttpRequestBody is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request body via tflog.
+	FieldHttpRequestBody = "tf_http_req_body"
+
+	// FieldHttpResponseProtoVersion is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response protocol version via tflog.
+	FieldHttpResponseProtoVersion = "tf_http_res_version"
+
+	// FieldHttpResponseStatusCode is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response status code via tflog.
+	FieldHttpResponseStatusCode = "tf_http_res_status_code"
+
+	// FieldHttpResponseStatusReason is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response status reason phrase via tflog.
+	FieldHttpResponseStatusReason = "tf_http_res_status_reason"
+
+	// FieldHttpResponseBody is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response body via tflog.
+	FieldHttpResponseBody = "tf_http_res_body"
+
+	// MessageHttpRequest is the message used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request via tflog.
+	MessageHttpRequest = "Sending HTTP Request"
+
+	// MessageHttpResponse is the message used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response via tflog.
+	MessageHttpResponse = "Received HTTP Response"
+)
+
+type loggingHttpTransport struct {
+	subsystem string
+	transport http.RoundTripper
+}
+
+func (t *loggingHttpTransport) RoundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
Taking another look at https://github.com/hashicorp/terraform-plugin-sdk/issues/546, I think we may want to consider one more thing there to cover the ask: generating our own "HTTP request ID". Having a separately generated ID means you can peek at a request or response, copy that ID value, then find anything related to that specific HTTP request/response.

This might be something like the `tf_req_id` log field, which is managed upstream by terraform-plugin-go to automatically generate a UUID value that gets persisted with the logger contexts for each RPC request/response. [Upstream code reference](https://github.com/hashicorp/terraform-plugin-go/blob/4a24b5d5377f223ca890cdada0b6fdbab7410d45/internal/logging/context.go#L58-L71).

I think we can separately decide whether we should be setting our own User-Agent if none is configured, so we can add that generated ID into there are well.

Reply:
I think it makes sense to add that UUID. But are you asking me to use the one set by `terraform-plugin-go` or to add a new UUID, generated for each Req/Res pair that goes through the Transport?

Reply:
To clarify: I'm asking because each request, comes in with a `Context` and we have documented (and will add it to the website doc), that they need to pass in the context they receive as part of the RPC callbacks into the provider code.

So, my assumption is, that UUID is... already there?

Reply:
Sorry I wasn't clear 😅 I meant generate a new UUID for each HTTP request, separate from the other one. Each RPC may result in multiple HTTP requests so its good to provide separate IDs for each. 👍

Reply:
Ah, great.

Yeah, totally happy to do it, if that closes #546 properly: seems like a long overdue ask 👍